drivers/dma/dma_stm32_v3.c

```diff
@@ -0,0 +1,763 @@
+/*
+ * Copyright (C) 2025 Savoir-faire Linux, Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "dma_stm32_v3.h"
+
+#include <zephyr/drivers/clock_control.h>
+#include <zephyr/drivers/dma/dma_stm32.h>
+
+#include <zephyr/irq.h>
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(dma_stm32_v3, CONFIG_DMA_LOG_LEVEL);
+
+#define DT_DRV_COMPAT st_stm32_dma_v3
+#define DMA_STM32_MAX_DATA_ITEMS 0xffff
+
+/* Since the descriptors pool is allocated statically, we define the number of
+ * descriptors per channel to be used for linked list trasnfers.
+ */
+#define CONFIG_DMA_STM32_NUM_DESCRIPTORS_PER_CHANNEL 24
+
+static const uint32_t table_src_size[4] = {
```

Comment:
Nit: Omit size if it is a magic number
```suggestion
static const uint32_t table_src_size[] = {
```

ditto two arrays below